### PR TITLE
samples: ipc: rpmsg_service: Add support for stm32h747i_disco

### PR DESCRIPTION
**Pull request by celinakalus**
With the changes made in pull request #68741, RPMsg service is supported on stm32h747i_disco using the STM32 HSEM IPM driver. For the sample to work, add device tree overlays to enable the mailbox and set the shared memory appropriately.

### CODE REVIEW

File: samples/subsys/ipc/rpmsg_service/README.rst [hunk 1]
```diff
@@ -128,3 +128,35 @@ remote) will appear on the corresponding serial ports:
...
+
+    west build -d build-m7/ -b stm32h747i_disco/stm32h747xx/m7 .
+    west build -d build-m4/ -b stm32h747i_disco/stm32h747xx/m4 remote/
+
+Then, you can flash both cores of the disco board like so:
+
+.. code-block:: console
+
+    west flash -d build-m7/
+    west flash -d build-m4/
```
Comment (nordicjm) [inline samples/subsys/ipc/rpmsg_service/README.rst:147]: does not use the correct doc template for building

File: samples/subsys/ipc/rpmsg_service/README.rst [hunk 2]
```diff
@@ -128,3 +128,35 @@ remote) will appear on the corresponding serial ports:
...
+
+Building the application for stm32h747i_disco/stm32h7xx
+*******************************************************
+
+Using west, build the sample for both cores separately like so:
+
+.. code-block:: console
+
+    west build -d build-m7/ -b stm32h747i_disco/stm32h747xx/m7 .
+    west build -d build-m4/ -b stm32h747i_disco/stm32h747xx/m4 remote/
```
Comment (nordicjm) [inline samples/subsys/ipc/rpmsg_service/README.rst:140]: and instead of this, use sysbuild, so it will build both images

**Review by nordicjm — CHANGES_REQUESTED** [review]
Add additional cmake/kconfig code so that building this using sysbuild will automatically build both images

File: samples/subsys/ipc/rpmsg_service/README.rst (hunk 2 above)
Comment (celinakalus) [reply samples/subsys/ipc/rpmsg_service/README.rst]: done

File: samples/subsys/ipc/rpmsg_service/README.rst (hunk 1 above)
Comment (celinakalus) [reply samples/subsys/ipc/rpmsg_service/README.rst]: fixed

**Review by nordicjm — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Pull request merged**